### PR TITLE
Update sandbox documentation - allow all pan

### PR DESCRIPTION
**Pull request by Lassejoe**
Update the documentation after allowing all PANs https://github.com/clearhaus/edss-go/pull/267.

The new trigger is the last 4 in PAN which is documented here.
I have also updated some of the text to make it bit more readable.

There is a section with examples that tries to help out integrators getting started.

### CODE REVIEW

**Review by cbobach — COMMENTED** [review]
Just some comments:
1. Are the first column necessary in the tables?
2. What happens if one is to provide a digit not specified? Should we add a catch all/default in the bottom of all the tables, and what should this then be?
3. In the ARes table the 4th digit is set to 3 on most of the test cases this seems to collide with the rest of the changes? I would have expected an `X` at this point.

Otherwise a really good job done :clap:

**Comment by Lassejoe** [comment]
> 1. Are the first column necessary in the tables?

I personally think it improves the readability, but maybe it is a personal preference and if so, I'm not against to remove it.

> 2. What happens if one is to provide a digit not specified? Should we add a catch all/default in the bottom of all the tables, and what should this then be?

They will receive an error where the description is the following:
`"Unknown test, see valid tests here https://docs.3dsecure.io/3dsv2/sandbox.html"`


> 3. In the ARes table the 4th digit is set to 3 on most of the test cases this seems to collide with the rest of the changes? I would have expected an X at this point.

This was necessary when creating the test cases. So if it is a frictionless flow the 4th digit has to be `3`.

**Comment by cbobach** [comment]
> I personally think it improves the readability, but maybe it is a personal preference and if so, I'm not against to remove it.

It's  fine by me :smile: 

> They will receive an error where the description is the following:
"Unknown test, see valid tests here https://docs.3dsecure.io/3dsv2/sandbox.html"

Should we state this somewhere on the page?

> This was necessary when creating the test cases. So if it is a frictionless flow the 4th digit has to be 3.

I would probably have flipped the 3th and 4th then such that the 3th digit determined the flow type friction-less or challenge, and then transStatus as the 4th :shrug:

Something in the line off:

### Challenge type

| Third digit | PAN last 4 | Description   | Requirements                          |
| ----------- | ---------- | ------------- | ------------------------------------- |
| 0           | xx0x       | Friction-less | transaction status is required        |
| 7           | xx7x       | Challenge     | invocation of `/postauth` is required |

### Transaction status

| Fourth digit | PAN last 4 | Description   | Requirements           |
| ------------ | ---------- | ------------- | ---------------------- |
| 0            | xxx0       | transStatus Y |                        |
| 1            | xxx1       | transStatus N |                        |
| 2            | xxx2       | transStatus A | [^1] |
| 3            | xxx3       | transStatus R | [^1] |
| 4            | xxx4       | transStatus I | [^1] |
| 5            | xxx5       | transStatus U | [^1] |
| 6            | xxx6       | DS timeout    | [^1] |
| 7            | xxx7       | Manual challange with transStatus Y or N | [^2] |

[^1]: Only applicable for challenge type: Friction-less, last 4: `xx0x`.
[^2]: Only applicable for challenge type: Challenge, last 4: `xx7x`.

Just a thought :shrug:

**Comment by Lassejoe** [comment]
> Should we state this somewhere on the page?

We can write something a la this:
_If the last four digits do not match any of the given test cases above, an error will be given_

**Comment by Lassejoe** [comment]
> I would probably have flipped the 3th and 4th then such that the 3th digit determined the flow type friction-less or challenge, and then transStatus as the 4th 🤷

I see your point but not all of the transStatus matches if it is a challenge flow. So it will become messy, IMO.

**Comment by cbobach** [comment]
> We can write something a la this:
> If the last four digits do not match any of the given test cases above, an error will be given

I think that would be nice :bow

File: source/sandbox.rst [hunk 1]
```diff
@@ -202,9 +202,15 @@ Read :ref:`postauth usage <postauth-usage>` for understanding how to fetch chall
       - Manual challenge with `transStatus` `Y` or `N`
       - `transStatus` `C` in `ARes` see `ARes outcome`_
 
-********
+*****
+Error
+*****
+
+If the last four digits do not match any of the given test cases above, an error will be given.
```
Comment (cbobach) [inline source/sandbox.rst:209]: I believe that this is only the case if a specific BIN range is used, right? As now only some ranges are ending towards our `standin` :shrug:

Comment (Lassejoe) [reply source/sandbox.rst:209]: All ranges are towards our standin.

**Review by cbobach — APPROVED** [review]
:muscle:

**Pull request merged**